packages/netlify-cms-widget-list/src/ListControl.js

```diff
@@ -310,8 +311,14 @@ export default class ListControl extends React.Component {
   objectLabel(item) {
     const { field } = this.props;
     if (this.getValueType() === valueTypes.MIXED) {
-      return getTypedFieldForValue(field, item).get('label', field.get('name'));
+      const itemType = getTypedFieldForValue(field, item);
+      const fallbackLabel = itemType.get('label');
+      const summaryTemplate = itemType.get('summary');
+      return summaryTemplate
+        ? compileStringTemplate(summaryTemplate, null, '', item)
```

Comment:
We'll want to handle Date in an item's content here. Also we may want to pass in the type's label, so user can do i.e `{{type.label}} | {{name}}` instead of manually copying the label into the string template

Reply:
You can add additional templates by modifying the item data, e.g.
```
item = item.set('label', itemType.get('label'))
```

Then `{{label}}` should work.
See: https://github.com/netlify/netlify-cms/blob/f88c83b97a18d9094240b688739d8b2909f19b0c/packages/netlify-cms-core/src/lib/formatters.ts#L125

Reply:
If you have an existing `label` field you should be able to access it via `fields.label`

Reply:
Thanks @erezrokah, that's neat!